core/src/main/java/net/sourceforge/jnlp/util/WindowsDesktopEntry.java

```diff
@@ -145,51 +143,44 @@ public void createWindowsMenu() throws IOException {
-        if (ManageMode.A == mode) {
-            List<String> lines = null;
-            // if UTF-8 fails, try ISO-8859-1
-            try {
-                LOG.debug("Reading Shortcuts with UTF-8");
-                lines = Files.readAllLines(getWindowsShortcutsFile().toPath(), UTF_8);
-            } catch (MalformedInputException me) {
-                LOG.debug("Fallback to reading Shortcuts with default encoding {}", Charset.defaultCharset().name());
-                lines = Files.readAllLines(getWindowsShortcutsFile().toPath(), Charset.defaultCharset());
-            }
-            Iterator it = lines.iterator();
-            String sItem = "";
-            String sPath;
-            Boolean fAdd = true;
-            // check to see if line exists, if not add it
-            while (it.hasNext()) {
-                sItem = it.next().toString();
-                String[] sArray = sItem.split(",");
-                String application = sArray[0]; //??
-                sPath = sArray[1];
-                if (sPath.equalsIgnoreCase(path)) {
-                    // it exists don't add
-                    fAdd = false;
-                    break;
-                }
-            }
-            if (fAdd) {
-                LOG.debug("Default encoding is {}", Charset.defaultCharset().name());
-                LOG.debug("Adding Shortcut to list = {} with UTF-8 encoding", sItem);
-                String scInfo = file.getFileLocation().toString() + ",";
-                scInfo += path + "\r\n";
-                Files.write(getWindowsShortcutsFile().toPath(), scInfo.getBytes(UTF_8), StandardOpenOption.APPEND);
-            }
+        LOG.debug("Using WindowsShortCutManager {}", shortcutFile);
+        final List<String> lines = readAllLine(shortcutFile);
+        if (needToAddNewShortcutEntry(path, lines)) {
+            final String scInfo = file.getFileLocation().toString() + "," + path;
+            LOG.debug("Adding Shortcut to list: {}", scInfo);
+            lines.add(scInfo);
+            final String content = String.join("\r\n", lines);
+            FileUtils.saveFileUtf8(content, shortcutFile);
         }
     }
 
+    private List<String> readAllLine(File shortcutFile) throws IOException {
+        try {
+            LOG.debug("Reading Shortcuts with UTF-8");
+            return Files.readAllLines(shortcutFile.toPath(), UTF_8);
+        } catch (MalformedInputException me) {
+            LOG.debug("Fallback to reading Shortcuts with default encoding {}", Charset.defaultCharset().name());
+            return Files.readAllLines(shortcutFile.toPath(), Charset.defaultCharset());
+        }
+    }
+
+    private boolean needToAddNewShortcutEntry(String path, List<String> lines) {
+        return lines.stream()
+                .map(line -> line.split(","))
+                .map(array -> array[1])
+                .anyMatch(sPath -> sPath.equalsIgnoreCase(path));
```

Comment:
noneMatch - if we don't find a match then we need to add

Reply:
thanks, fixed